src/clj_kondo/impl/linters/keys.clj

```diff
@@ -2,18 +2,64 @@
   {:no-doc true}
   (:require
    [clj-kondo.impl.findings :as findings]
-   [clj-kondo.impl.utils :refer [node->line tag]]))
+   [clj-kondo.impl.utils :refer [node->line tag]]
+   [clojure.string :as string]))
+
+(defn- ensure-no-nils
```

Comment:
This is in case a collection contains a node we don't support as a key. Let me know if this defensive programming isn't necessary.

Reply:
Added a comment in the PR diff.

I think the defensive programming is correct here, but it can be made more efficient:

- By shortcutting on the first nil key-value
- By checking for nil before coercing into a set or map instead of walking that set or map later on

Reply:
Addressed in the latest commit.